code/modules/tables/tables.dm

```diff
@@ -130,6 +130,16 @@
 		var/obj/item/hand/H = W
 		if(H.cards && H.cards.len == 1)
 			usr.visible_message("\The [user] plays \the [H.cards[1].name].")
+
+	if(istype(W, /obj/item/deck)) //playing cards
+		if(user.a_intent == I_GRAB)
```

Comment:
Wait, why?

Reply:
Every attackby will trigger this. We make it check grab intent so it won't call deal proc.